tor/helpers/reddit_ids.py

```diff
@@ -37,3 +42,31 @@ def is_valid(post_id, config):
     # successfully and it's good to work on. If the insert fails, then we
     # want to return a False because we cannot work on that post again.
     return add_complete_post_id(post_id, config, return_result=True)
+
+
+def is_removed(post: Submission) -> bool:
+    """
+    Reddit does not allow non-mods to tell whether or not a post has been
+    removed, which understandably makes it a little difficult to figure out
+    whether or not we should remove a post automatically. HOWEVER, as with
+    pretty much everything else that Reddit does, they left enough detritus
+    that we guess with a relatively high degree of accuracy.
+
+    We can use a combination of checking to see if something is gildable or
+    crosspostable (and its current content, if a selfpost) to identify its
+    current state:
+
+    flag                    can_gild    is_crosspostable    selftext
+    removed                 Y           N                   ""
+    dead user               N           Y                   ""
+    removed + dead user     N           N                   ""
+    deleted post            N           N                   [deleted]
+
+    More information (and where this table is from): https://redd.it/7hfnew
+
+    :param post: The post that we are attempting to investigate.
+    :param config: the global config object.
+    :return: True is the post looks like it has been removed, False otherwise.
+    """
+
+    return False if post.is_crosspostable and post.can_gild else True
```

Comment:
This is great! But I don't see a reason to remove posts if only the author has deleted their account, so we shouldn't have to check `can_gild`:  `return not post.is_crosspostable` should do.